File: Week 1/C++ Codes/FizzBuzz.cpp
#include<bits/stdc++.h>
#include<iostream>
using namespace std;

void fizzBuzz1(int n)
{
    for(int i = 1; i<=n; i++)
    {
        int flag = 0;
        if (i%3 == 0)
        {
            flag = 1;
            cout<<"Fizz";
        }
        if (i%5 == 0)
        {
            flag = 1;
            cout<<"Buzz";
        }
        if (flag == 0)
            cout<<i;
        cout<<"\n";
    }
}

void fizzBuzz2(int n)
{
    for(int i = 1; i<=n; i++)
    {
        int flag = 0;
        if (i%15 == 0)
            cout<<"FizzBuzz";
        else if (i%3 == 0)
            cout<<"Fizz";
        else if (i%5 == 0)
            cout<<"Buzz";
        else
            cout<<i;
        cout<<"\n";
    }
}

int main()
{
    fizzBuzz1(120);
    fizzBuzz2(120);
    return 0;
}

File: Week 1/Python Codes/FizzBuzz.py
# Brute Force Implementation
def FizzBuzz1(n):
    for i in range(1, n+1):
        if i%15 == 0:
            print("FizzBuzz", end="")
        elif i%3 == 0:
            print("Fizz", end="")
        elif i%5 == 0:
            print("Buzz", end="")
        else:
            print(i, end="")
        print()

# Efficient Implementation
def FizzBuzz2(n):
    for i in range(1, n+1):
        flag = 0
        if i%3 == 0:
            print("Fizz", end="")
            flag = 1
        if i%5 == 0:
            print("Buzz", end="")
            flag = 1
        if flag == 0:
            print(i, end="")
        print()

if __name__ == "__main__":
#    n = int(input())
    n = 10
    FizzBuzz1(n)
    FizzBuzz2(n)
        